src/testers.jl

```diff
@@ -123,48 +127,40 @@ function test_scalar(f, z; rtol=1e-9, atol=1e-9, fdm=_fdm, fkwargs=NamedTuple(),
     Δx = one(z)
     @testset "$f at $z, with tangent $Δx" begin
         # check ∂u_∂x and (if Ω is complex) ∂v_∂x via forward mode
-        frule_test(f, (z, Δx); rtol=rtol, atol=atol, fdm=fdm, fkwargs=fkwargs, kwargs...)
+        frule_test(f, (z, Δx); rule_test_kwargs...)
         if z isa Complex
             # check that same tangent is produced for tangent 1.0 and 1.0 + 0.0im
-            @test isapprox(
-                frule((Zero(), real(Δx)), f, z; fkwargs...)[2],
-                frule((Zero(), Δx), f, z; fkwargs...)[2],
-                rtol=rtol,
-                atol=atol,
-                kwargs...,
-            )
+            _, real_tangent = frule((Zero(), real(Δx)), f, z; fkwargs...)
+            _, embedded_tangent = frule((Zero(), Δx), f, z; fkwargs...)
+            _check_equal(real_tangent, embedded_tangent; isapprox_kwargs...)
         end
     end
     if z isa Complex
```

Comment:
I want to move the 4 sections of this function that happen only for complex out into 2 helper functions.
(one for forward on one for reverse)